### PR TITLE
Doc fixes

### PR DESCRIPTION
**Pull request by richsalz**
A handful of things.  Some real errors, some style.  I think we should avoid the word "legacy" except when talking about old clients or servers.

not worth backporting.

### CODE REVIEW

File: doc/man3/SSL_set_bio.pod [hunk 1]
```diff
@@ -33,55 +33,62 @@ functions.
 SSL_set_bio() does a similar job as SSL_set0_rbio() and SSL_set0_wbio() except
 that it connects both the B<rbio> and the B<wbio> at the same time. This
 function transfers the ownership of B<rbio> and B<wbio> to B<ssl> except that
-the rules for this are much more complex. For this reason this function is
-considered a legacy function and SSL_set0_rbio() and SSL_set0_wbio() should be
-used in preference. The ownership rules are as follows:
+the rules for this are much more complex and are defined as follows:
+The ownership rules are as follows:
```
Comment (mattcaswell) [inline doc/man3/SSL_set_bio.pod]: You now have two lines that end in "as follows:". Why are you dropping the advice to use SSL_set0_rbio() and SSL_set0_wbio() in preference?

File: doc/man3/PEM_read_bio_PrivateKey.pod [hunk 2]
```diff
@@ -170,8 +170,8 @@ EVP_PKEY structure. The write routines use PKCS#8 private key format and are
 equivalent to PEM_write_bio_PKCS8PrivateKey().The read functions transparently
 handle traditional and PKCS#8 format encrypted and unencrypted keys.
 
-PEM_write_bio_PrivateKey_traditional() writes out a private key in legacy
-"traditional" format.
+PEM_write_bio_PrivateKey_traditional() writes out a private key in the
+"traditional" format with a simple private key marker.
```
Comment (mattcaswell) [inline doc/man3/PEM_read_bio_PrivateKey.pod]: To me the word "legacy" here was implying that this format shouldn't normally be used unless you need it. That doesn't come across in the new wording.

File: doc/man3/SSL_set_bio.pod (hunk 1 above)
Comment (mattcaswell) [inline doc/man3/SSL_set_bio.pod]: Oh - I see you added a note at the end. I think that note should be at the beginning somewhere.

File: doc/man3/X509_NAME_print_ex.pod [hunk 3]
```diff
@@ -35,10 +35,11 @@ characters. Multiple lines are used if the output (including indent) exceeds
...
 =head1 NOTES
 
-The functions X509_NAME_oneline() and X509_NAME_print() are legacy functions which
+The functions X509_NAME_oneline() and X509_NAME_print()
 produce a non standard output form, they don't handle multi character fields and
-have various quirks and inconsistencies. Their use is strongly discouraged in new
-applications.
+have various quirks and inconsistencies.
+Their use is strongly discouraged in new applications and they will
+be deprecated in a future release.
```
Comment (mattcaswell) [inline doc/man3/X509_NAME_print_ex.pod]: Should they be marked as deprecated for 1.2.0 in the headers?

File: doc/man3/SSL_set_bio.pod (hunk 1 above)
Comment (richsalz) [inline doc/man3/SSL_set_bio.pod]: I think it is okay as it is, and the flow of the documentation reads better this way.

File: doc/man3/X509_NAME_print_ex.pod (hunk 3 above)
Comment (richsalz) [inline doc/man3/X509_NAME_print_ex.pod]: I am in favor of that, but this PR is just for documentation.  I left us "wiggle room" to do it whenever we want.  In fact I changed it to say "could be deprecated"

**Comment by richsalz** [comment]
Additional commit pushed to address comments so far.

**Comment by richsalz** [comment]
I added another commit fixing 1.1 to 1.1.0, please re-review.

**Comment by richsalz** [comment]
Thanks very much for the feedback!  Squashed and merged.

**Pull request closed without merging**